playground/playground.go

```diff
@@ -174,6 +210,38 @@ func main() {
 			scope.Set("code", string(out))
 			scope.Set("output", []Line{})
 		})
+
+		scope.Set("share", func() {
+			req := js.Global.Get("XMLHttpRequest").New()
+			req.Call("open", "POST", "http://"+snippetStoreHost+"/share", true)
+			req.Set("responseType", "arraybuffer")
+			req.Set("onload", func() {
+				if req.Get("status").Int() != 200 {
+					scope.Apply(func() {
+						scope.Set("output", []Line{Line{"type": "err", "content": `cannot share snippet`}})
+					})
+					return
+				}
+
+				data := js.Global.Get("Uint8Array").New(req.Get("response")).Interface().([]byte)
+				scope.Apply(func() {
+					id := string(data)
+
+					dom.GetWindow().Location().Hash = "#/" + id
+
+					scope.Set("shareUrl", dom.GetWindow().Location().Str())
+					scope.Set("showShareUrl", true)
+					// TODO: Do this better using AngularJS.
+					//       Perhaps using http://stackoverflow.com/questions/14833326/how-to-set-focus-on-input-field/18295416.
+					go func() {
+						time.Sleep(time.Millisecond)
+						dom.GetWindow().Document().GetElementByID("share-url").(*dom.HTMLInputElement).Select()
+					}()
+				})
+			})
+			// TODO: Should be sending as binary blob or is text okay?
```

Comment:
Actually, I'm not completely sure if it causes issues, it might've been due to something else.